File: tandem_bicycle/tandem_bicycle.py
def tandem_bicycle(red_shirt_speeds, blue_shirt_speeds, fastest):
  if fastest == True:
    red_shirt_speeds.sort(reverse=True)
    blue_shirt_speeds.sort()
  else:
    red_shirt_speeds.sort()
    blue_shirt_speeds.sort()
  
  total = 0

  for idx in range(len(blue_shirt_speeds)):
    blue_speed = blue_shirt_speeds[idx]
    red_speed = red_shirt_speeds[idx]
    total += max(blue_speed, red_speed)

  return total

red_shirt_speeds = [5, 5, 3, 9, 2]
blue_shirt_speeds = [3, 6, 7, 2, 1]

print(tandem_bicycle(red_shirt_speeds, blue_shirt_speeds, True))
print(tandem_bicycle(red_shirt_speeds, blue_shirt_speeds, False))

File: tandem_bicycle/tandem_bicycle.go
package main

import (
    "fmt"
    "sort"
)

func tandemBicycle(redShirtSpeeds, blueShirtSpeeds []int, fastest bool) int {
    sort.Ints(redShirtSpeeds)
    sort.Ints(blueShirtSpeeds)

    if fastest {
        reverse(redShirtSpeeds)
    }

    total := 0
    for i := 0; i < len(redShirtSpeeds); i++ {
        if redShirtSpeeds[i] > blueShirtSpeeds[i] {
            total += redShirtSpeeds[i]
        } else {
            total += blueShirtSpeeds[i]
        }
    }
    return total
}

func reverse(arr []int) {
    for i, j := 0, len(arr)-1; i < j; i, j = i+1, j-1 {
        arr[i], arr[j] = arr[j], arr[i]
    }
}

func main() {
    red := []int{5, 5, 3, 9, 2}
    blue := []int{3, 6, 7, 2, 1}
    fmt.Println(tandemBicycle(red, blue, true))
    fmt.Println(tandemBicycle(red, blue, false))
}